hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/RecordStore.java

```diff
@@ -314,5 +310,18 @@
      **/
     void maybeDoInitialLoad();
 
+    Storage createStorage(RecordFactory<R> recordFactory, InMemoryFormat memoryFormat);
+
+    Record createRecord(Object value, long ttlMillis, long now);
+
+    /**
+     * This can be used to release unused resources.
+     */
+    void dispose();
+
+    void destroy();
+
+    Storage getStorage();
```

Comment:
Now you are exposing the underlying abstraction. I would not do this. If someone needs to do something on the Storage, just add a method on the RecordStore. If you get a lot of such pass through calls, then probably the abstraction is broken.